components/SingleArticle.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { fetchArticleById } from '../apis';
+
+
+const SingleArticle = ({article}) => {
+  const [currentArticle, setCurrentArticle] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const { id } = useParams();
+  
+  useEffect(() => {
+    setIsLoading(true);
+    fetchArticleById(id)
+    .then(({article}) => {
+      setCurrentArticle(article);
+      setIsLoading(false);
+    })
+  }, [id])
+  
+  const { article_id, topic, title, author, created_at, votes, article_img_url, comment_count, body } = currentArticle;
+
+  if (isLoading) {
+    return (
+      <>
+        <p>Loading article...</p>
+      </>
+    )
+  }
+  else {
+    return (
+      <div className= "single-article">
```

Comment:
Is there a more semantic parent element that could be used here? Something other than a div?